src/services/blocks/BlocksService.ts

```diff
@@ -524,10 +575,52 @@ export class BlocksService extends AbstractService {
 			specName,
 			specVersion,
 			decorated,
-			runtimeDoesNotMatch,
 		};
 	}
 
+	/**
+	 * Extracts necessary `extrinsicBaseWeight` or `blockWeights` data from
+	 * the decorated metadata in order to cache memory efficiently.
+	 *
+	 * @param decorated Decorated meta data to be parsed
+	 * @param runtimeVersion spec version of the blocks runtime
+	 * @returns
+	 */
+	private extractWeightFromDecorated(
+		decorated: DecoratedMeta,
```

Comment:
I wonder if this should be `decorated.consts.system` and not the a pointer to the entire `DecorateMeta` object. I think that would make sense and allow for a testable `extractWeightFromDecorated`

Reply:
FWIU, your suggestion makes sense